packages/quantic/force-app/main/default/lwc/quanticResultLink/quanticResultLink.js

```diff
@@ -90,10 +99,47 @@ export default class QuanticResultLink extends LightningElement {
     );
   };
 
+  checkResultType() {
+    if (this.result?.raw?.sfid !== undefined) {
+      this.isSalesforceLink = true;
+    }
+  }
+
+  handleOpenTabSalesforceLink(event) {
+    event.stopPropagation();
+
+    const targetPageRef = {
+      type: 'standard__recordPage',
+      attributes: {
+        recordId: this.getRecordIdAttribute(),
+        objectApiName: this.getObjectApiNameAttribute(),
+        actionName: 'view',
+      },
+    };
+    this[NavigationMixin.Navigate](targetPageRef);
```

Comment:
Is there a case where this operation fails? like if the recordId or the objectApiName are incorrect or missing for example.
We should fallback the usual behaviour of the QuanticResultLink component if this operation fails.

Reply:
addressed [here](https://github.com/coveo/ui-kit/pull/2804#:~:text=ok%20I%20will,href%3D%7Bresult.clickUri%7D)